components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -606,17 +608,40 @@ export default class PluginManagement extends AdminSettings {
         return this.installFromUrl(true);
     }
 
+    handleRemovePluginCancel = () => {
+        this.setState({
+            showRemoveModal: false,
+        });
+        this.resolveRemoveModal(false);
+    }
+
+    handleRemovePlugin = () => {
+        this.setState({showRemoveModal: false});
+        this.resolveRemoveModal(true);
+    }
+
+    showRemovePluginModal = () => {
+        this.setState({showRemoveModal: true});
+        return new Promise((res) => {
+            this.resolveRemoveModal = res;
+        });
+    }
+
     handleRemove = async (e) => {
         this.setState({lastMessage: null, serverError: null});
         e.preventDefault();
         const pluginId = e.currentTarget.getAttribute('data-plugin-id');
-        this.setState({removing: pluginId});
 
-        const {error} = await this.props.actions.removePlugin(pluginId);
-        this.setState({removing: null});
+        const result = await this.showRemovePluginModal();
```

Comment:
We can have this be called directly by the button here: https://github.com/mattermost/mattermost-webapp/blob/a6074923f8ac42f3c058b1801a7f3ca98358783a/components/admin_console/plugin_management/plugin_management.jsx#L793